Define and demonstrate data transform types within documentation (#142)
cytomining/CytoTable: docs/source/overview.md

```{eval-rst}
  Parquet data destination type may be specified by using :code:`convert(..., dest_datatype="parquet", ...)` (:mod:`convert() <cytotable.convert.convert>`).
```

## Data Transformations

CytoTable performs various types of data transformations.
This section help define terminology and expectations surrounding the use of this terminology.

> One or all of these may be used by CytoTable depending on the way it is configured.

Reviewer: ```suggestion
CytoTable might use one or all of these depending on user configuration.
```

Reviewer: (active voice)

Author: Thanks!

### Data Chunking

<table>
<tr><th>Original</th><th>Changes</th></tr>
<tr>
<td>

"Data source"

<table>
<tr><th>Col_A</th><th>Col_B</th><th>Col_C</th></tr>
<tr><td>1</td><td>a</td><td>0.01</td></tr>
<tr><td>2</td><td>b</td><td>0.02</td></tr>
</table>

</td>
<td>

"Chunk 1"

<table>
<tr><th>Col_A</th><th>Col_B</th><th>Col_C</th></tr>
<tr><td>1</td><td>a</td><td>0.01</td></tr>

</table>

"Chunk 2"

<table>
<tr><th>Col_A</th><th>Col_B</th><th>Col_C</th></tr>
<tr><td>2</td><td>b</td><td>0.02</td></tr>
</table>

</td>
</tr>
</table>

_Example of data chunking performed on a simple table of data._

Data chunking within CytoTable involves slicing data sources into "chunks" of rows which all contain the same columns and have a lower number of rows than the original data source.

> Data chunking is generally used to reduce the memory footprint of operations performed within CytoTable.

Reviewer: ```suggestion
CytoTable uses data chunking to reduce the memory footprint of operations.
```

Active voice

Author: Thank you for calling this out!

> Data chunking can be used through [Parquet "datasets"](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetDataset.html), which are one form of serialized and chunked data.

Reviewer: I'm a bit confused by the "can be used" terminology. It can be used if a user sets a certain configuration? Or is it default?

Pending the answer to this question, please also update to active voice.

Author: Thanks, I'll try to clarify! Chunking might not apply (for example, if the chunksize is as big or larger than the full dataset row count). Within the context of Parquet datasets, I think I should reword for clarity - as this is more about implementation (both within CytoTable or by the user outside of CytoTable) than strictly data chunking itself.

Author: Added further clarity here with documentation and chunk_size argument reference.

### Data Concatenations

<table>
<tr><th>Original</th><th>Changes</th></tr>
<tr>
<td>

"Chunk 1"

<table>
<tr><th>Col_A</th><th>Col_B</th><th>Col_C</th></tr>
<tr><td>1</td><td>a</td><td>0.01</td></tr>

</table>

"Chunk 2"

<table>
<tr><th>Col_A</th><th>Col_B</th><th>Col_C</th></tr>
<tr><td>2</td><td>b</td><td>0.02</td></tr>
</table>

</td>
<td>

"Concatenated data"

<table>
<tr><th>Col_A</th><th>Col_B</th><th>Col_C</th></tr>
<tr><td>1</td><td>a</td><td>0.01</td></tr>
<tr><td>2</td><td>b</td><td>0.02</td></tr>
</table>

</td>
</tr>
</table>

_Example of data concatenation performed on simple tables of similar data "chunks"._

> Data concatenation within CytoTable involves appending two or more data sources (for example, as chunks from above) with the same columns together as a unified dataset.

Reviewer: You're introducing the term "appending" which is a bit confusing. Maybe the example makes it clear?

Author: Thanks - I'll try to find other words here to describe what's happening.

When rendered, the example table(s) looks like this:

![image](https://github.com/cytomining/CytoTable/assets/3738008/f136d6b2-40c6-4ef3-b780-c048d5f8ce95)

Author: Removed the word "appending" and tried to use "chunks" as a reference point instead.

Join Specification in SQL
</td>
</tr>
<tr >
<td colspan="2">

```sql
SELECT *
FROM Table_1
LEFT JOIN Table_2 ON
Table_1.Col_A = Table_2.Col_A;
```

</td>
</tr>
</table>

_Example of a data join performed on simple example tables._

```{eval-rst}
Data joins within CytoTable involve bringing one or more data sources together with differing columns as a new dataset.
The word "join" here is interpreted through `SQL-based terminology on joins <https://en.wikipedia.org/wiki/Join_(SQL)>`_.
Joins may be specified in CytoTable using `DuckDB-style SQL <https://duckdb.org/docs/sql/introduction.html>`_ through :code:`convert(..., joins="SELECT * FROM ... JOIN ...", ...)` (:mod:`convert() <cytotable.convert.convert>`).
Also see CytoTable's presets found here: :data:`presets.config <cytotable.presets.config>` or via `GitHub source code for presets.config <https://github.com/cytomining/CytoTable/blob/main/cytotable/presets.py>`_.
```

Note: data software outside of CytoTable sometimes makes use of the term "merge" to describe capabilities which are similar to join (for ex. [`pandas.DataFrame.merge`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.merge.html).

> Within CytoTable, we opt to describe these operations with "join" to avoid confusion with software development alongside the technologies used (for ex. [DuckDB SQL](https://duckdb.org/docs/archive/0.9.2/sql/introduction) includes no `MERGE` keyword).

Reviewer: ```suggestion
Within CytoTable, we opt to describe these operations with "join" to avoid confusion with software development alongside the technologies used (for example, [DuckDB SQL](https://duckdb.org/docs/archive/0.9.2/sql/introduction) includes no `MERGE` keyword).
```
